### ISSUE TITLE
1 second lag

### ISSUE THREAD

**Issue by mkhakpaki** [BrikerMan/BMPlayer#70]
Thank you for your great framework it helped me a lot.
but there is a bug with slider and current time label 
current time and slider position is 1 second ago of real current time even on sample project 
as my videos duration are short it is a big bug for me 
for example when video ends slider stops at 14/15 and steel is 1/15 away of final end position
how can i fix this?

**Comment by BrikerMan** [comment]
I am using the `BMTimeSlider` to fix the position error in the original Slider, maybe it is not that perfect. You can try to fix this~ 

```swift
open class BMTimeSlider: UISlider {
    override open func trackRect(forBounds bounds: CGRect) -> CGRect {
        let trackHeigt:CGFloat = 2
        let position = CGPoint(x: 0 , y: 14)
        let customBounds = CGRect(origin: position, size: CGSize(width: bounds.size.width, height: trackHeigt))
        super.trackRect(forBounds: customBounds)
        return customBounds
    }
    
    override open func thumbRect(forBounds bounds: CGRect, trackRect rect: CGRect, value: Float) -> CGRect {
        let rect = super.thumbRect(forBounds: bounds, trackRect: rect, value: value)
        let newx = rect.origin.x - 10
        let newRect = CGRect(x: newx, y: 0, width: 30, height: 30)
        return newRect
    }
}

```

**Comment by mkhakpaki** [comment]
is it a solution?
if it is, how can i use that ? 
where to add these codes?
thanks

**Comment by BrikerMan** [comment]
No, this is just the Code of slider which I make changed. 
You should change the source code of `BMTimeSlider `. 
This is why it is not perfect and how you fix this, I have to work on #68 first.
1. https://codedump.io/share/z2k9wcxuOukC/1/uislider-thumb-image-doesn39t-start-from-the-beginning
2. http://stackoverflow.com/questions/35306248/uislider-not-taking-up-full-width
